### PR TITLE
Product Rating block: Fix review count not displaying on frontend

### LINKED ISSUE #8970: Product Reviews: Reviews counter not displaying
## Describe the bug

The review counter for the Product Reviews block is not being displayed when the block is inside the Single Product block. The number of reviews should appears to the right of the stars, inside parenthesis, such as: `(1 customer review)`

## To reproduce

1. Log in to your WordPress dashboard;
2. Go to Appearance > Themes, and select a block theme (for example: Twenty-twenty three);
3. Go to Appearance > Editor;
4. Click the Edit button;
5. Click on the "+" icon to add a new block and search for "Single Product" block in the search bar;
6. Click on the "Single Product" block to add it to your page or post;
7. Select a product on the Product Selector and click on the Save button;
8. On the frontend, visit a product that contains at least 1 review and check that the counter is not displayed;

## Screenshots

| Current | Expected |
--- | ---
| image | image |

### WordPress (please complete the following information):

-   WordPress version: 6.2
-   Gutenberg version (if installed): 15.5.0
-   WooCommerce version: 7.6.0-dev
-   WooCommerce Blocks version: 10.0.0-dev

### Desktop (please complete the following information):

-   OS: MacOS Ventura
-   Browser Chrome
-   Version 111.0.5563.146

### PR DESCRIPTION
**Pull request by thealexandrelara**
The review counter for the Product Reviews block is not being displayed when the block is inside the Single Product block. The number of reviews should appears to the right of the stars, inside parenthesis, such as: (1 customer review)

Fixes #8970 

### <samp>🤖 Generated by Copilot at 2e6543c</samp>

*  Refactor `Block` component in `block.tsx` to use hooks for data and context, and add new props for different scenarios ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9129/files?diff=unified&w=0#diff-56b0c48092fcb0fd507a289ea3b56a01607e951d78f516a38d30f656b625e9f6L23-L27), [link](https://github.com/woocommerce/woocommerce-blocks/pull/9129/files?diff=unified&w=0#diff-56b0c48092fcb0fd507a289ea3b56a01607e951d78f516a38d30f656b625e9f6L112-R138), [link](https://github.com/woocommerce/woocommerce-blocks/pull/9129/files?diff=unified&w=0#diff-56b0c48092fcb0fd507a289ea3b56a01607e951d78f516a38d30f656b625e9f6L160-R181))
* Update `style.scss` to reflect new class name for single product block and add flexbox styling for rating container ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9129/files?diff=unified&w=0#diff-0ec89a253e28931bdfdbb528c1b9f187e66dad78d2b325059234e94075294732L61-R69))
* Add `parse_attributes` method to `ProductRating` class in `ProductRating.php` to handle default values and parsing of block attributes ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9129/files?diff=unified&w=0#diff-9defe615963571610affb73341a43db364eeb7fc5616000176494510be3e8e70R55-R72))
* Replace `filter_rating_html` method with anonymous function and pass new attributes to filter rating HTML in `render_callback` method in `ProductRating.php` ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9129/files?diff=unified&w=0#diff-9defe615963571610affb73341a43db364eeb7fc5616000176494510be3e8e70L72-L93), [link](https://github.com/woocommerce/woocommerce-blocks/pull/9129/files?diff=unified&w=0#diff-9defe615963571610affb73341a43db364eeb7fc5616000176494510be3e8e70L112-R180))

---

#### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

### Screenshots

| Before | After |
| ------ | ----- |
| image | image |

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Log in to your WordPress dashboard;
2. Go to Appearance > Themes, and select a block theme (for example: Twenty-twenty three);
3. Go to Appearance > Editor;
4. Click the Edit button;
5. Click on the "+" icon to add a new block and search for "Single Product" block in the search bar;
6. Click on the "Single Product" block to add it to your page or post;
7. Select a product that contains at least 1 review on the Product Selector and click on the Save button;
8. On the editor, check that the Review count is displayed;
9. Go to the frontend of your website;
10. On the frontend, check that the Review count is displayed

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental

### Changelog

> Add reviews count to the Product Rating block

### CODE REVIEW

**Review by gigitux — CHANGES_REQUESTED** [review]
LGTM the code!

What should the behavior be when the user clicks `Add Review`? Currently, when the user clicks, the URL is updated by adding the suffix #reviews, but nothing happens on the page. I think that we should fix it.

https://user-images.githubusercontent.com/4463174/234520861-48572d25-8c80-45d0-b91d-2d45b00f3d7e.mp4

**Comment by thealexandrelara** [comment]
> LGTM the code!
> 
> What should the behavior be when the user clicks `Add Review`? Currently, when the user clicks, the URL is updated by adding the suffix #reviews, but nothing happens on the page. I think that we should fix it.
> 
>  Screen.Capture.on.2023-04-26.at.10-43-01.1.mp4

Good point, I think we should fix this as well. I think that currently there is no dedicated page for reviews, so there is no place to redirect the user. Considering that, I can see 2 possible behaviors:
1. Hide the "Add review" when there is no review for the selected product;
2. Just display empty stars, like: ☆☆☆☆☆ 

I just created [an issue for that](https://github.com/woocommerce/woocommerce-blocks/issues/9270), so I can work on this in a separate PR, and we can avoid blocking this one until we decide which way would be better. What do you think?

**Comment by gigitux** [comment]
Reading the comment that @kmanijak did (https://github.com/woocommerce/woocommerce-blocks/issues/9270#issuecomment-1527057997). I think that we should render the new HTML only when the block is used inside the Single Product Block and the Single Product template. What do you think?

File: src/BlockTypes/ProductRating.php [hunk 1]
```diff
@@ -109,13 +105,77 @@ protected function render( $attributes, $content, $block ) {
...
+						'<div class="wc-block-components-product-rating__container">
+							<div class="wc-block-components-product-rating__stars wc-block-grid__product-rating__stars" role="img" aria-label="%1$s">
+								%2$s
+							</div>
+							%3$s
+						</div>
+						',
+						esc_attr( $label ),
+						wc_get_star_rating_html( $average_rating, $reviews_count ),
+						$is_descendent_of_single_product_block ? $reviews_count_html : ''
```
Comment (gigitux) [inline src/BlockTypes/ProductRating.php:166]: I think that it is necessary to update this logic. We have to render the `$reviews_count_html`, when the block is used inside the Single Product Template too.

image

**Review by gigitux — APPROVED** [review]
Reading the @kmanijak's comment (https://github.com/woocommerce/woocommerce-blocks/issues/9270#issuecomment-1527057997), I think that we should add the support for the Single Product Template. I created a dedicated issue for this #9287.

**Comment by thealexandrelara** [comment]
> Reading the @kmanijak's comment ([#9270 (comment)](https://github.com/woocommerce/woocommerce-blocks/issues/9270#issuecomment-1527057997)), I think that we should add the support for the Single Product Template. I created a dedicated issue for this #9287.

Ohh nice, we can work in a separate issue then, I'll be merging this one so it will be easier to start from a new branch. Thank you for reviewing this one 🙌

**Pull request merged**